experimenter/experimenter/experiments/migrations/0266_nimbusexperiment_results_pairwise_data.py

```diff
@@ -0,0 +1,165 @@
+            #                                     "weekly": {},
+            #                                     "overall: {}
+            #                                 }
+            #                             },
+            #                             "percent": 0.0
+
+            comparison_default = {"first": {}, "all": []}
+            significance_default = {"weekly": {}, "overall": []}
+
+            for key, value in data["v3"].items():
+                if value is None or key not in ["weekly", "overall"]:
+                    continue
+                for basis, basis_data in value.items():
+                    for segment, segment_data in basis_data.items():
+                        # look up reference_branch from results data if we don't already know it
+                        if reference_branch is None:
+                            for branch_name, v in segment_data.items():
+                                if v["is_control"] == True:
+                                    reference_branch = branch_name
+                                    break
+                        for cur_branch, branch_data_obj in segment_data.items():
+                            branch_data = branch_data_obj["branch_data"]
+                            for (
+                                metrics_group,
+                                metrics,
+                            ) in branch_data.items():
+                                for metric, metric_data in metrics.items():
+                                    for (
+                                        comparison,
+                                        comparison_data,
+                                    ) in metric_data.items():
+                                        if not isinstance(comparison_data, dict):
+                                            continue
+                                        metric_data_keys = [
+                                            k for k in comparison_data.keys()
+                                        ]
+                                        if not set(branches).issubset(metric_data_keys):
+                                            comparison_data_value = deepcopy(
+                                                comparison_data
+                                            )
+                                            if comparison in [
+                                                "difference",
+                                                "relative_uplift",
+                                            ]:
+                                                data["v3"][key][basis][segment][
+                                                    cur_branch
+                                                ]["branch_data"][metrics_group][metric][
+                                                    comparison
+                                                ] = {}
```

Comment:
We do this deeply nested indexing repeatedly. Lets pull some of these out into vars so its more readable.